Window Covering devices

ebaauw opened the pull request:
Update support for Aqara Roller Shade Driver E1 (`lumi.curtain.acn002`), see #5330:
- Fix `type`: now `Window covering device` instead of `Dimmable light`;
- Add write-only `state` attribute `lift_inc` step up (-1), stop (0), and step down (1).  Values < -1 are mapped to -1, values > 1 to 1;
- Use _Multistate Output_ cluster for `stop`, `open`, and `lift_inc`;
- Add write-only `state` attribute `alert` to blink the led;
- Add read/write `state` attribute `speed` to set the motor speed: 0 = low, 1 = medium, 2 = high.
- Handle additional attributes of Lumi-specific cluster.  Created a separate attribute set for this device;
- Handle additional tags for Xiaomi special attribute;
- Create ZHABattery `/sensors` resource to expose the battery and charging state.

Update support for Aqara Motor B1 (`lumi.curtain.hagl04`), see #4249:
- Use _Multistate Output_ cluster for `stop`, `open`, and `close`;
- Handle additional tags for Xiaomi special attribute;
- Bug fix: B1 can no longer be controlled through the API;
- Expose charging state on ZHABattery resource (as yet untested).

Update support for IKEA FYRTUR and KADRILJ:
- Add write-only `state` attribute `alert` to blink the led.

~~Remove deprecated `state` attributes `on`, `bri`, `bri_inc`, and `sat` from window covering devices.  Use `open`, `lift`, `stop`, and `tilt` instead, see https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2475#issuecomment-609485739.~~

Remove workarounds for `deCONZ::ZclAttribute` bug handling `deCONZ::ZclSingleFloat` and `deCONZ::Zcl8BitEnum` values.

Update _Node List_ panel in the GUI when receiving new values for _Basic_ cluster attributes.

Added another Sunricher wireless switch, `ZG2833K4_EU06,` to `button_maps.json`.

Added manufacturer-specific attributes for Sunricher ZG9092 thermostat to `general.xml`, see #5409.

Bug fixes:
- Wrong API code when specifying an invalid value for `alert` when PUTting a lights state.
- Cannot set `sunriseoffset` nor `sunsetoffset` in `config` for built-in Daylight sensor.

Smanar commented:
Some code part are needed to make the device usable, but there is some hacks possible to remove if we use DDF too no ?

ebaauw commented:
I think the PR is good to go.  It's running on my production network (now on bullseye).  I verified it for the Aqara Roller Shade Driver E1, the IKEA FYRTUR, and the original mains-powered Aqara Curtain Motor.

Attention points:
- There's a discussion on the **#technical-talk** channel in Discord on the removal of the deprecated `state` attributes.  Thomas indicates this will cause issues with Hue Essentials.  If needed, I can revert the corresponding commit.
- I ditched the SOMA Tilt in favour of the Aqara E1 for my Venetian blinds.  I added support in Homebridge Hue [v0.13.34](https://github.com/ebaauw/homebridge-hue/releases/tag/v0.13.34) for using the E1 with Venetian blinds.  If this is something other API clients would find useful, we might consider adding this functionality to the API instead.
- I had an extensive session online with @Krocko analysing the Aqara Curtain Motor B1, but this PR hasn't yet been tested with this device.  In particular I don't know whether `state.charging` for the ZHABattery will work.
- There are still missing event notifications when _Basic_ cluster attributes change.  Also, I'm not sure if the GUI is updated in all cases, but I fixed a few missing cases I spotted.
- The E1 doesn't expose _SW Build ID_, but reports the firmware version in the Xiaomi special attribute.  However, `swversion` is updated from _Date Code_ as well as from _SW Build ID_.  The REST attribute reflects the last Zigbee attribute read.  I think _Date Code_ shouldn't override _SW Build ID_.  Per the ZCL standard this should actually reflect the production date of the hardware, not of the firmware!

ebaauw commented:
> Some code part are needed to make the device usable, but there is some hacks possible to remove if we use DDF too no ?
> I speak during the device inclusion (creation in the API)

I would love to remove hacks, as you know.  Especially those needed because the `/lights` resource is created before the _Basic_ cluster has been read.  I don't know how to handle this using DDF, though.

Smanar commented:
>I  would love to remove hacks, as you know. Especially those needed because the /lights resource is created before the Basic cluster has been read

Yeah was exactly those one I am thinking, but I have some problem too on my side with device created with DDF but managed with legacy code.

EloB commented:
I’m like a baby boy waiting for Santa on this PR. Do you guys have any estimate when this will be released? 🙂

PalmMx commented:
Just curious, what is happening with this PR?
Almost a month now since the last action.
I think we are many end-users waiting for this to be launched. 😃

Mimiix commented:
Please read: https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5733

PalmMx commented:
> Please read: https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5733
> 
> 

OK, I see. But what does this mean in time? Will it take a week or two, or are we talking several months? A best guess would be sufficient.

Mimiix commented:
I _Hope_ a week or 2.

I know that @ebaauw his PR's are decent, so i can ask Manup to give this priority.

@PalmMx Are you running deCONZ on an native OS or use Docker/HA Addon? If you run uit native, you can compile the changes yourself and run it that way.

PalmMx commented:
> I _Hope_ a week or 2.
> 
> I know that @ebaauw his PR's are decent, so i can ask Manup to give this priority.
> 
> @PalmMx Are you running deCONZ on an native OS or use Docker/HA Addon? If you run uit native, you can compile the changes yourself and run it that way.

Priority to this would really be appreciated.

I'm running deCONZ on a RPi with Raspbian (Raspbian Buster Desktop). Is there any way to compile the changes for a semi-advanced user like myself, please let me know how.

Thanks

PalmMx commented:
Is this close to being launched now? 😃

ebaauw commented on rest_lights.cpp:
```diff
@@ -1948,7 +1967,6 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
                 taskRef.lightNode->modelId() == QLatin1String("Shutter SW with level control"))
             {
                 taskRef.lightNode->setValue(RStateLift, 50);
-                taskRef.lightNode->setValue(RStateBri, 127);
```
Also here
